client/extensions/woocommerce/components/product-image-uploader/index.js

```diff
@@ -0,0 +1,219 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+import { head, uniqueId, find, noop } from 'lodash';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { errorNotice as errorNoticeAction } from 'state/notices/actions';
+
+import DropZone from 'components/drop-zone';
+import FilePicker from 'components/file-picker';
+import MediaActions from 'lib/media/actions';
+import MediaUtils from 'lib/media/utils';
+import MediaStore from 'lib/media/store';
+import MediaValidationStore from 'lib/media/validation-store';
+
+class ProductImageUploader extends Component {
+	static propTypes = {
+		siteId: PropTypes.number,
+		multiple: PropTypes.bool,
+		compact: PropTypes.bool,
+		onSelect: PropTypes.func,
+		onUpload: PropTypes.func,
+		onError: PropTypes.func,
+		onFinish: PropTypes.func,
+	};
+
+	static defaultProps = {
+		multiple: true,
+		compact: false,
+		onSelect: noop,
+		onUpload: noop,
+		onError: noop,
+		onFinish: noop,
+	}
+
+	state = {
+		errors: [],
+	};
+
+	showError = ( media, transientId ) => {
+		const { onError, errorNotice, translate } = this.props;
+		const { errors } = this.state;
+
+		onError( {
+			file: media,
+			transientId,
+		} );
+
+		let extraDetails;
+		const validationError = errors[ transientId ] || [];
+		switch ( head( validationError ) ) {
+			case 'EXCEEDS_MAX_UPLOAD_SIZE' :
+				extraDetails = translate( 'This file exceeds the maximum upload size.' );
+				break;
+			case 'EXCEEDS_PLAN_STORAGE_LIMIT' :
+			case 'NOT_ENOUGH_SPACE' :
+				extraDetails = translate( 'You have reached your plan storage limit.' );
+				break;
+		}
+
+		const message = translate( 'There was a problem uploading %s.', {
```

Comment:
Not sure this is the best idea re: localization. It is possible "your image", below, without context, won't translate well into the sentence here. May I recommend splitting this out into two strings + logic instead.